### ISSUE TITLE
Efficiency of training data generation

### ISSUE THREAD

**Issue by cycao77** [megvii-research/NeurIPS2021-ML4CO-KIDA#1]
hello, i found that the efficiency of training data generation is a big problem when running your codes.  Although multi-threading is used here, as far as I know, python multi-threading is not much faster than single threading. So, i want to know if you have any other methods to speed up the generation of training data?  Thank you！

**Comment by caozixuan** [comment]
1) During the ML4CO competition, because we have enough time and computing power, the amount of training data we use is very **redundant** and we keep our competition settings in this repo. Especially in Item Placement, reducing the size of training and validation data in each round by **10 times** may also get good results.  For Anonymous, perhaps reducing the data size and iteration rounds is also feasible.

2) As for multi-threading, our data collection code refers to https://github.com/ds4dm/ml4co-competition/blob/main/baseline/dual/train_files/01_generate_dataset.py . The process of collecting data involves much I / O. According to our observation, the running efficiency of multithreading is high.  If you have sufficient computing resources, you can open more processes to collect data and use multiple GPUs during training.

**Comment by cycao77** [comment]
> 1. During the ML4CO competition, because we have enough time and computing power, the amount of training data we use is very **redundant** and we keep our competition settings in this repo. Especially in Item Placement, reducing the size of training and validation data in each round by **10 times** may also get good results.  For Anonymous, perhaps reducing the data size and iteration rounds is also feasible.
> 2. As for multi-threading, our data collection code refers to https://github.com/ds4dm/ml4co-competition/blob/main/baseline/dual/train_files/01_generate_dataset.py . The process of collecting data involves much I / O. According to our observation, the running efficiency of multithreading is high.  If you have sufficient computing resources, you can open more processes to collect data and use multiple GPUs during training.

Thank you for your reply!

**Comment by cycao77** [comment]
Now, I have another problem. 
I have trained the 50 best model parameters using your code, but I don't know how to choose the best 3 out of them. Best acc or minimal loss? Hope to get your answer, thank you very much!

**Comment by caozixuan** [comment]
We find that the loss and accuracy are not strongly related to the rewards to be evaluated. We evaluate the rewards of 10, 20, 30, 40, 50 iterations and choose models that have top rewards. As for calculating rewards, you can refer to our submission directory and the documentation of ML4CO.
